MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,364 @@
+
+Saving an entity:
+```java
+public void save(SampleEntity entity) {
+    mongoOperations.save(entity);
+```
+
+Find by id:
+```java
+public Optional<SampleEntity> findById(String id) {
+    return Optional.ofNullable(mongoOperations.findById(id, SampleEntity.class));
+```
+
+More complex queries can be realised by building a compound `Query`:
+```java
+public Optional<SampleEntity> findByArbitraryFields(String identifier, String value) {
+  return Optional.ofNullable(
+    mongoOperations.findOne(
+      new Query(
+        where(EntityFields.ARBITRARY_FIELD).is(identifier))
+          .and(EntityFields.OTHER_FIELD).ne(value),
+        SampleEntity.class));
+}
+```
+See the [official documentation](https://docs.spring.io/spring-data/mongodb/docs/current/api/org/springframework/data/mongodb/core/query/Criteria.html) for further information.
+
+#### Auto-generate queries out of method names with the `MongoRepository` interface
+Depending on the complexity, queries can also be auto-generated based on the method names by using
+the MongoRepository Interface provided by spring-data-mongo.
+
+Imagine the following simple entity class:
+```java
+public class Person {
+  private String Name;
+  private int age;
+}
+```
+
+Now we can define queries directly through the method names (see the official documentation for the specific syntax).
+```java
+public interface PersonMongoRepository extends MongoRepository<Person, String> {
+  Optional<Person> findByName(String name);
+
+  List<Person> findAllByAgeIsLessThanEqual(int age);
+
+  List<Person> findAllByNameIsNot(String name);
+}
+```
+See the [official documentation](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details) for further details.
+
+#### Query functions
+
+As Spring Data Mongo doesn't support/provide many query functions provided by Morphia, below are some replacements/alternatives.
+
+##### EqualIgnoreCase
+
+Morphia supports usage of _equalIgnoreCase()_. Use _regex()_ in Spring Data Mongo. For example
+
+Morphia - `query.criteria("fieldName").equalIgnoreCase(entity.getFieldName());`
+
+Spring Data Mongo - `query.addCriteria(where("fieldName").regex("^" + entity.getFieldName(), "i"));`
+
+##### HasAnyOf
+
+Morphia supports _hasAnyOf()_ method. Use _in()_ in Spring Data Mongo. For example 
+
+Morphia - `query.field("id").hasAnyOf(getIds());`
+
+Spring Data Mongo - `query.addCriteria(where("id").in(getIds()));`
+
+##### Filter
+
+Morphia supports _filter()_ method. Use _is()_ in Spring Data Mongo. For example
+
+Morphia - `query.filter("id", getId());`
+
+Spring Data Mongo - `query.addCriteria(where("id").is(getId()));`
+
+##### Contains
+
+Morphia supports _contains()_ method. Use _regex()_ in Spring Data Mongo. For example
+
+Morphia - `query.criteria("fieldName").contains(entity.getFieldName());`
+
+Spring Data Mongo - `query.addCriteria(where("fieldName").regex(".*" + entity.getFieldName() + ".*"));`
```

Comment:
I'd use `Pattern.quote` ([see explanation](https://www.baeldung.com/java-pattern-quote#use-patternquote)) to make readers aware that this might be necessary.

Reply:
updated the query. Let me know if it required further explanation